reaper/task.go

```diff
@@ -0,0 +1,259 @@
+package reaper
```

Comment:
**hint**: Incidental - healthier to have each reaper in a separate file. Straight copy.